File: src/main/java/ru/anarcom/octopus/model/BaseEntity.java
package ru.anarcom.octopus.model;

import java.util.Date;
import javax.persistence.Column;
import javax.persistence.EnumType;
import javax.persistence.Enumerated;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.MappedSuperclass;
import lombok.Data;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedDate;

/**
 * Base class with property 'id'.
 * Used as a base class for all objects that requires this property.
 *
 * @author Eugene Suleimanov
 * @version 1.0
 */

@MappedSuperclass
@Data
public class BaseEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    public Long id;

    @CreatedDate
    @Column(name = "created")
    public Date created;

    @LastModifiedDate
    @Column(name = "updated")
    public Date updated;

    @Enumerated(EnumType.STRING)
    @Column(name = "status")
    public Status status;
}

File: src/main/kotlin/ru/anarcom/octopus/entity/BaseEntity.kt
package ru.anarcom.octopus.entity

import org.springframework.data.annotation.CreatedDate
import org.springframework.data.annotation.LastModifiedDate
import java.time.Instant
import javax.persistence.*

/**
 * Base class for Entity with ID, created and updated fields.
 */
@MappedSuperclass
abstract class BaseEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null

    @CreatedDate
    @Column(name = "created")
    var created: Instant? = null

    @LastModifiedDate
    @Column(name = "updated")
    var updated: Instant? = null

    @Enumerated(EnumType.STRING)
    @Column(name = "status")
    var status: Status? = null
}
